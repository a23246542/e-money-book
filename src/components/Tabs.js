import React from 'react';
import PropTypes from 'prop-types';

const Tabs = ({ children, activeIndex, onTabChange }) => {
  const selectTab = (e, index) => {
    e.preventDefault();
    onTabChange(index);
  };
  return (
    <ul className="nav nav-tabs nav-justified" data-testid="navTabs">
      {React.Children.map(children, (item, index) => {
        const activeClassName =
          index === activeIndex ? 'nav-link active' : 'nav-link';
        return (
          <li className="nav-item" data-test="navItem">
            <a
              className={activeClassName}
              onClick={(e) => {
                selectTab(e, index);
              }}
              role="button"
            >
              {item}
            </a>
          </li>
        );
      })}
    </ul>
  );
};

const Tab = ({ children }) => {
  return <React.Fragment>{children}</React.Fragment>;
};

Tabs.propTypes = {
  activeIndex: PropTypes.number.isRequired,
  onTabChange: PropTypes.func.isRequired,
};

export { Tabs, Tab };
