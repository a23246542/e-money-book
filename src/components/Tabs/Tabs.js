import React from 'react';
import PropTypes from 'prop-types';
import style from './style.module.scss';

export const Tabs = ({ children, activeIndex, onTabChange }) => {
  const selectTab = (e, index) => {
    e.preventDefault();
    onTabChange(index);
  };
  return (
    <ul
      className={`${style['tabs']} nav nav-tabs nav-justified`}
      data-testid="navTabs"
    >
      {React.Children.map(children, (item, index) => {
        const activeClassName =
          index === activeIndex ? 'nav-link active' : 'nav-link';
        return (
          <li className="nav-item" data-test="tabItem">
            <a
              className={activeClassName}
              onClick={(e) => {
                selectTab(e, index);
              }}
              data-testid={`tab-${index}`}
              role="button"
            >
              {item}
            </a>
          </li>
        );
      })}
    </ul>
  );
};

export const Tab = ({ children }) => {
  return <React.Fragment>{children}</React.Fragment>;
};

Tabs.propTypes = {
  activeIndex: PropTypes.number.isRequired,
  onTabChange: PropTypes.func.isRequired,
};
