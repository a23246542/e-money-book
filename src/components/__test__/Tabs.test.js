import React from 'react';
import { shallow, mount } from 'enzyme';
import { Tabs, Tab} from '../Tabs';

const props = {
  activeIndex:0,
  onTabChange:jest.fn()
};
let wrapper = {};

describe('test Tabs and Tab component', ()=>{
  //進行每個最小用例之前，重新shallow
  beforeEach(()=>{
    wrapper = shallow(
      <Tabs {...props}>
        <Tab>1</Tab>
        <Tab>2</Tab>
      </Tabs>
    )
  })
  //渲染組件要符合快照
  it('should render the component to match the snapshot',()=>{
    expect(wrapper).toMatchSnapshot();
  })
  //默認渲染兩個tab，第一個標籤要是active

  //交互點第二個tab會改變active狀態跟觸發對的函式
})