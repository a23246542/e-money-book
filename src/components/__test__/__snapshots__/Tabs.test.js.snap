// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Tabs and Tab component should render the component to match the snapshot 1`] = `ReactWrapper {}`;
